src/Controls/src/Core/Page/Page.cs

```diff
@@ -346,30 +458,53 @@ protected override void OnBindingContextChanged()
 				SetInheritedBindingContext(_titleView, BindingContext);
 		}
 
+		/// <summary>
+		/// Indicates that the preferred size of a child <see cref="Element"/> has changed.
+		/// </summary>
+		/// <param name="sender">The object that raised the event.</param>
+		/// <param name="e">The event arguments.</param>
 		protected virtual void OnChildMeasureInvalidated(object sender, EventArgs e)
 		{
 			InvalidationTrigger trigger = (e as InvalidationEventArgs)?.Trigger ?? InvalidationTrigger.Undefined;
 			OnChildMeasureInvalidated((VisualElement)sender, trigger);
 		}
 
+		/// <summary>
+		/// When overridden, allows the application developer to customize behavior as the page disappears.
+		/// </summary>
+		/// <remarks>
+		/// This method is called when the page disappears due to navigating away from the page within the app. It is <b>not</b> called when the app disappears due to an event external to the app (e.g. user navigates to the home screen or another app, a phone call is received, the device is locked, the device is turned off).
+		/// </remarks>
 		protected virtual void OnDisappearing()
 		{
 		}
 
+		/// <summary>
+		/// Called when the page's <see cref="Element.Parent"/> property has changed.
+		/// </summary>
+		/// <exception cref="InvalidOperationException">Thrown when the parent is not a (descendant of) <see cref="Page"/> or <see cref="BaseShellItem"/>.</exception>
```

Comment:
I'm a bit confused by this statement. I would expect things to break when the parent _**is**_ a descendant of the page/shell, because (in my mind) a parent element shouldn't really be a child element of the page itself. How does this work?

Reply:
I think it's trying to say that the `Parent` of a `Page` must be castable to `Page` or `BaseShellItem`.

Reply:
No worries! Maybe exactly what EZ said? 
```suggestion
		/// <exception cref="InvalidOperationException">Thrown when the page's <see cref="RealParent"/> can't be casted to <see cref="Page"/> or <see cref="BaseShellItem"/>.</exception>
```